modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp

```diff
@@ -772,202 +810,316 @@ void RenderableFieldlinesSequence::setModelDependentConstants() {
+                fls::Model::Invalid,
+                _scalingFactor
+            );
         }
+    }
+    catch (const std::runtime_error& e) {
+        LERROR(e.what());
+    }
+}
 
-        if (outVec.empty()) {
-            LERROR(std::format("Found no seed points in '{}'", seedFilePath));
-            outMap.clear();
-            return {};
+void RenderableFieldlinesSequence::trackOldest(File& file) {
+    if (file.status == File::FileStatus::Loaded) {
+        std::deque<File*>::iterator it =
+            std::find(_loadedFiles.begin(), _loadedFiles.end(), &file);
+        if (it == _loadedFiles.end()) {
+            _loadedFiles.push_back(&file);
+        }
+        // Repopulate the queue if new File makes the queue full
+        if (!_loadedFiles.empty() &&
+            _loadingType == LoadingType::DynamicDownloading &&
+            _loadedFiles.size() >= _maxLoadedFiles)
+        {
+            File* oldest = _loadedFiles.front();
+            // The edge case of when queue just got full and user jumped back to where
+            // they started which would make the oldes file in queue to be the active
+            // file. In that case we need to make sure we do not unload it
+            if (oldest == &file) {
+                return;
+            }
+            oldest->status = File::FileStatus::Downloaded;
+            oldest->state.clear();
+            _loadedFiles.pop_front();
         }
+    }
+}
 
-        size_t lastIndex = seedFilePath.find_last_of('.');
-        std::string name = seedFilePath.substr(0, lastIndex);   // remove file extention
-        size_t dateAndTimeSeperator = name.find_last_of('_');
-        std::string time = name.substr(dateAndTimeSeperator + 1, name.length());
-        std::string date = name.substr(dateAndTimeSeperator - 8, 8);    // 8 for yyyymmdd
-        std::string dateAndTime = date + time;
+int RenderableFieldlinesSequence::updateActiveIndex(double currentTime) {
+    if (_files.empty()) {
+        return -1;
+    }
+    // Sets index to 0 if time is at the first file time stamp and also,
+    // if size == 1 then we can expect to not have a sequence and wants to show
+    // the one file of fieldlines at all points in time
+    if (_files.begin()->timestamp == currentTime || _files.size() == 1) {
+        return 0;
+    }
+
+    int index = 0;
+    const std::deque<File>::const_iterator iter = std::upper_bound(
+        _files.begin(),
+        _files.end(),
+        currentTime,
+        [](double timeRef, const File& fileRef) {
+            return timeRef < fileRef.timestamp;
+        }
+    );
 
-        // add outVec as value and time stamp as int as key
-        outMap[dateAndTime] = outVec;
+    if (iter == _files.begin()) {
+        index = 0;
     }
-    return outMap;
+    else if (iter != _files.end()) {
+        index = static_cast<int>(std::distance(_files.cbegin(), iter));
+    }
+    else {
+        index = static_cast<int>(_files.size()) - 1;
+    }
+    return index;
```

Comment:
You could remove the `index` variable and just return in each if-statement instead

```suggestion
    else if (iter != _files.end()) {
        return static_cast<int>(std::distance(_files.cbegin(), iter));
    }
    else {
        return static_cast<int>(_files.size()) - 1;
    }
``` 
also in the first if above :)